src/panels/DocumentTabV2.ts

```diff
@@ -0,0 +1,178 @@
+    ) {
+        super(panel, DocumentTab.viewType, { hasConnection: connection ? 'true' : 'false' });
+
+        DocumentTab.openTabs.add(this);
+
+        this.connection = connection;
+        this.documentId = documentId ?? undefined;
+        this.mode = mode;
+
+        if (connection.masterKey) {
+            this.telemetryContext.addMaskedValue(connection.masterKey);
+        }
+
+        this.telemetryContext.addMaskedValue(connection.databaseId);
+        this.telemetryContext.addMaskedValue(connection.containerId);
+
+        this.session = new DocumentSession(connection, this.channel);
+    }
+
+    public static render(
+        connection: NoSqlQueryConnection,
+        mode: DocumentTabMode,
+        documentId?: CosmosDbRecordIdentifier,
+        viewColumn?: vscode.ViewColumn,
+    ): DocumentTab {
+        const column = viewColumn ?? vscode.ViewColumn.One;
+        if (documentId) {
+            const openTab = [...DocumentTab.openTabs].find((openTab) => {
+                if (!openTab.documentId) {
+                    return false;
+                }
+
+                if (documentId._rid && openTab.documentId._rid && openTab.documentId._rid === documentId._rid) {
+                    return true;
+                }
+
+                if (documentId.partitionKey !== undefined && openTab.documentId.partitionKey !== undefined) {
+                    const openTabPK = Array.isArray(openTab.documentId.partitionKey)
+                        ? openTab.documentId.partitionKey.join(',')
+                        : openTab.documentId.partitionKey?.toString();
+                    const pk = Array.isArray(documentId.partitionKey)
+                        ? documentId.partitionKey.join(',')
+                        : documentId.partitionKey?.toString();
+
+                    return documentId.id === openTab.documentId.id && openTabPK === pk;
+                }
+
+                return documentId.id === openTab.documentId.id;
+            });
+
+            if (openTab) {
+                openTab.panel.reveal(column);
+                return openTab;
+            }
+        }
+
+        const title = `${documentId?.id ? documentId.id : 'New Document'}.json`;
+        const panel = vscode.window.createWebviewPanel(DocumentTab.viewType, title, column, {
+            enableScripts: true,
+            retainContextWhenHidden: true,
+        });
+
+        return new DocumentTab(panel, connection, mode, documentId);
+    }
+
+    public dispose(): void {
+        DocumentTab.openTabs.delete(this);
+
+        this.session.dispose();
+
+        super.dispose();
+    }
+
+    protected initController() {
+        super.initController();
+
+        this.channel.on<void>('ready', async () => {
+            await this.channel.postMessage({
+                type: 'event',
+                name: 'initState',
+                params: [this.mode, this.connection.databaseId, this.connection.containerId, this.documentId],
```

Comment:
This goes to the webview, right? As a future optimization, we could send this directly in the template.